### PR TITLE
[Catgirl] Add ability to use waifupics API

### PR DESCRIPTION
**Pull request by LPlamondon**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Added support for the catgirls cog to make use of waifu.pics sfw/neko. Admins can toggle the use of this
It makes use of the Python requests library, so this will need to be installed as well

### CODE REVIEW

File: cogs/catgirl/catgirl.py [hunk 1]
```diff
@@ -330,3 +361,12 @@ def getImage(imageList, title):
         embed.add_field(name="Info", value=image["character"], inline=False)
     embed.set_image(url=image[KEY_IMAGE_URL])
     return embed
+
+
+def getImageUrl(image):
```
Comment (Injabie3) [inline cogs/catgirl/catgirl.py:365]: can we add a docstring to this, similar to the one in `getImage`?

(in retrospect I realize how poorly I named the original one)

File: cogs/catgirl/catgirl.py [hunk 2]
```diff
@@ -286,6 +302,21 @@ async def add(self, ctx, link: str, description: str = ""):
         else:
             await ctx.send("Added, notified and pending approval. :ok_hand:")
 
+    # [p]nyaa toggle
+    @_nyaa.command(name="toggle")
+    @checks.admin_or_permissions(manage_guild=True)
```
Comment (Injabie3) [inline cogs/catgirl/catgirl.py]: let's set this to mods and above.
```suggestion
    @checks.mod_or_permissions(manage_guild=True)
```

File: cogs/catgirl/catgirl.py [hunk 3]
```diff
@@ -102,8 +106,20 @@ async def catgirlCmd(self, ctx):
         """Displays a random, cute catgirl :3"""
         # Send typing indicator, useful when Discord explicit filter is on.
         await ctx.channel.trigger_typing()
-
-        embed = getImage(self.catgirls, "Catgirl")
+        nekoToggle = await self.config.guild(ctx.guild).waifuneko()
+
+        if nekoToggle == True:
```
Comment (Injabie3) [inline cogs/catgirl/catgirl.py]:
```suggestion
        if nekoToggle:
```

File: cogs/catgirl/catgirl.py [hunk 4]
```diff
@@ -102,8 +106,20 @@ async def catgirlCmd(self, ctx):
...
+            choice = random.randint(0, 1)
+            if choice == 0:
+                embed = getImage(self.catgirls, "Catgirl")
+            else:
+                #
+                URL = "https://api.waifu.pics/sfw/neko"
```
Comment (Injabie3) [inline cogs/catgirl/catgirl.py]: nit: can you put this constant at the top with the rest of the constants?

File: .gitignore [hunk 5]
```diff
@@ -143,3 +143,8 @@ ENV/
 
 # Pre-commit hooks
 /.pre-commit-config.yaml
+cogs/Bank/settings.json
+cogs/catgirl/settings.json
+cogs/CogManager/settings.json
+cogs/ModLog/settings.json
+core/settings.json
```
Comment (Injabie3) [inline .gitignore]: can we remove the gitignore changes? cog configs shouldn't be within the source tree anyways.

**Review by Injabie3 — CHANGES_REQUESTED** [review]
Thanks! Overall LGTM. What are your thoughts on handling HTTP exceptions for the waifupic api? Maybe if we get an exception, we can fall back to local ones?

note to self: can look into aiohttp later since requests might  be blocking.

File: cogs/catgirl/catgirl.py (hunk 1 above)
Comment (LPlamondon) [reply cogs/catgirl/catgirl.py:365]: Added the docstring as requested

File: cogs/catgirl/catgirl.py (hunk 4 above)
Comment (LPlamondon) [reply cogs/catgirl/catgirl.py]: Moved it up top with the other constants

File: cogs/catgirl/catgirl.py [hunk 6]
```diff
@@ -102,8 +107,19 @@ async def catgirlCmd(self, ctx):
...
-
-        embed = getImage(self.catgirls, "Catgirl")
+        nekoToggle = await self.config.guild(ctx.guild).waifuneko()
+
+        if nekoToggle:
+            choice = random.randint(0, 1)
+            if choice == 0:
+                embed = getImage(self.catgirls, "Catgirl")
+            else:
+                #
```
Comment (Injabie3) [inline cogs/catgirl/catgirl.py]: empty comment? probably can remove.

File: cogs/catgirl/catgirl.py (hunk 1 above)
Comment (Injabie3) [reply cogs/catgirl/catgirl.py:365]: Can you please move this docstring underneath the function declaration, similar to how it's done elsewhere? For context, docstrings show up if you do `.__doc__` on the function.

**Review by Injabie3 — CHANGES_REQUESTED** [review]
Fix it, then Ship it!

File: cogs/catgirl/catgirl.py (hunk 6 above)
Comment (LPlamondon) [reply cogs/catgirl/catgirl.py]: Removed

File: cogs/catgirl/catgirl.py (hunk 1 above)
Comment (LPlamondon) [reply cogs/catgirl/catgirl.py:365]: Moved it underneath. Sorry about that

**Review by Injabie3 — APPROVED** [review]

**Pull request merged**